comp/forwarder/defaultforwarder/forwarder_health.go

```diff
@@ -206,10 +237,11 @@ func (fh *forwarderHealth) validateAPIKey(apiKey, domain string) (bool, error) {
 	}
 
 	fh.setAPIKeyStatus(apiKey, domain, &apiKeyUnexpectedStatusCode)
-	return false, fmt.Errorf("Unexpected response code from the apikey validation endpoint: %v", resp.StatusCode)
+	return false, fmt.Errorf("unexpected response code from the apikey validation endpoint: %v", resp.StatusCode)
 }
 
-func (fh *forwarderHealth) hasValidAPIKey() bool {
+// check if any of the endpoints have a valid apiKey, updating health state
+func (fh *forwarderHealth) checkValidAPIKey() bool {
```

Comment:
Renaming this method to make it more obvious that it updates state; it doesn't merely calculate and return a bool.